lib/rules/no-global-regexp-flag-in-query.ts

```diff
@@ -0,0 +1,83 @@
+import { TSESTree } from '@typescript-eslint/utils';
+
+import { createTestingLibraryRule } from '../create-testing-library-rule';
+import {
+  isMemberExpression,
+  isIdentifier,
+  isCallExpression,
+  isProperty,
+  isObjectExpression,
+} from '../node-utils';
+
+export const RULE_NAME = 'no-global-regexp-flag-in-query';
+export type MessageIds = 'noGlobalRegExpFlagInQuery';
+type Options = [];
+
+export default createTestingLibraryRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallow the use of the global RegExp flag (/g) in queries',
+      recommendedConfig: {
+        dom: 'error',
+        angular: 'error',
+        react: 'error',
+        vue: 'error',
+      },
+    },
+    messages: {
+      noGlobalRegExpFlagInQuery:
+        'Avoid using the global RegExp flag (/g) in queries',
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context, _, helpers) {
+    function lint(
+      regexpNode: TSESTree.Literal,
+      identifier: TSESTree.Identifier
+    ) {
+      if (helpers.isQuery(identifier)) {
+        context.report({
+          node: regexpNode,
+          messageId: 'noGlobalRegExpFlagInQuery',
+        });
+      }
+    }
+
+    return {
+      [`CallExpression[callee.type=MemberExpression] > Literal[regex.flags=/g/].arguments`](
+        node: TSESTree.Literal
+      ) {
+        if (
+          isCallExpression(node.parent) &&
+          isMemberExpression(node.parent.callee) &&
+          isIdentifier(node.parent.callee.property)
+        ) {
+          lint(node, node.parent.callee.property);
+        }
+      },
+      [`CallExpression[callee.type=Identifier] > Literal[regex.flags=/g/].arguments`](
```

Comment:
Any specific reason to select only those `CallExpression` whose callee is `Identifier` or `MemberExpression`?

Reply:
AFAIK these two are the ones that can invoke a method `getByText()` and `something.getByText()`.
To get `getByText()` to verify it's on of our methods, I had to traverse the AST to find that identifier.
Maybe that there's a better way though 😅 (I hope that there is one)

Reply:
Oh, I see. You should be able to traverse the AST to find the identifier with [`getDeepestIdentifierNode`](https://github.com/testing-library/eslint-plugin-testing-library/blob/7bc2b9c10b2b39ded5135295e9cfa1472211cbf9/lib/node-utils/index.ts#L340):

> Gets the deepest identifier node in the expression from a given node.
> 
> An example:
> Having `const a = rtl.within('foo').getByRole('button')`:
> if we call `getDeepestIdentifierNode` with `rtl` node,
> it will return `getByRole` identifier

Reply:
Actually, seeing this example, I just remembered some issues with chained `within` in another rule. Could you add an extra test to make sure the rule reports correctly a query chained with `within`?